### PR TITLE
fix(Onboarding): Send all users through system permissions flow

### PR DESCRIPTION
**Pull request by KaylaBrady**
### Summary

No ticket - hot fix addressing app store review feedback.

What is this PR for?

This PR removes the "Not Now" / deferred location permissions flow in favor of sending all users to select their permissions through the system prompt.

- [X] If you added any user facing strings on iOS, are they included in Localizable.xcstrings?

### Testing

What testing have you done?
* Tested onboarding flow manually & with Voice Over on. Confirmed I could select my permission level & continue with teh onboarding flow automatically.
image

### CODE REVIEW

**Comment by KaylaBrady** [comment]
Fixing up some broken tests on this

File: iosApp/iosApp/Pages/Onboarding/OnboardingScreenView.swift [hunk 1]
```diff
@@ -227,24 +220,27 @@ struct OnboardingScreenView: View {
         .onReceive(inspection.notice) { inspection.visit(self, $0) }
     }
 
+    @ViewBuilder
+    var locationDescription: some View {
+        VStack {
+            Text("We use your location to show you nearby transit options.")
+                + Text("You can always change location settings later in the Settings app.")
```
[the remark this answers is not in the document]
Comment (KaylaBrady) [reply iosApp/iosApp/Pages/Onboarding/OnboardingScreenView.swift]: Updated to split these above / below the button based on the latest designs

File: iosApp/iosAppTests/Pages/Onboarding/OnboardingPageTests.swift [hunk 2]
```diff
@@ -16,21 +16,24 @@ final class OnboardingPageTests: XCTestCase {
...
+        let stepChannel = PassthroughSubject<Void, Never>()
+
         let sut = OnboardingPage(
             screens: OnboardingScreen.allCases,
             onFinish: { finishExp.fulfill() },
-            onboardingRepository: onboardingRepository
+            onAdvance: { stepChannel.send() },
+            onboardingRepository: onboardingRepository,
+            // Actual button location dialogue handling is unit tested in OnboardingScreenView.testLocationFlow
+            skipLocationDialogue: true
```
Comment (KaylaBrady) [inline iosApp/iosAppTests/Pages/Onboarding/OnboardingPageTests.swift:27]: I don't love this skip, but it seemed to be the most straightforward way to test.

File: iosApp/iosApp/Pages/Onboarding/OnboardingScreenView.swift [hunk 3]
```diff
@@ -165,21 +168,25 @@ struct OnboardingScreenView: View {
...
-                        Text("Allow Location Services").onboardingKeyButton()
-                    }
-                    Button(action: { shareLocation(false) }) {
-                        Text(
-                            "Skip for now",
-                            comment: "Button text for deferring the request for location services"
-                        ).onboardingSecondaryButton()
+                    Button(action: {
+                        // short circuit for OnboardingPageView integration testing
+                        if skipLocationDialogue {
```
Comment (EmmaSimon) [inline iosApp/iosApp/Pages/Onboarding/OnboardingScreenView.swift:180]: Non-blocking nitpick: I'd maybe move this into `shareLocation`, just for cleanliness.

File: iosApp/iosApp/Pages/Onboarding/OnboardingScreenView.swift [hunk 4]
```diff
@@ -176,12 +176,7 @@ struct OnboardingScreenView: View {
                         Spacer()
                     }
                     Button(action: {
-                        // short circuit for OnboardingPageView integration testing
-                        if skipLocationDialogue {
-                            advance()
-                        } else {
-                            shareLocation()
-                        }
+                        shareLocation()
```
Comment (EmmaSimon) [inline iosApp/iosApp/Pages/Onboarding/OnboardingScreenView.swift:179]: Oh, you should be able to just do `Button(action: shareLocation)` now.

**Pull request merged**